### ISSUE TITLE
Mermaid diagrams don't render inside closed Details elements

### ISSUE THREAD

**Issue by lampensau** [squidfunk/mkdocs-material#3559]
### Contribution guidelines

- [X] I've read the [contribution guidelines](https://github.com/squidfunk/mkdocs-material/blob/master/CONTRIBUTING.md) and wholeheartedly agree

### I've found a bug and checked that ...

- [X] ... the problem doesn't occur with the `mkdocs` or `readthedocs` themes
- [X] ... the problem persists when all overrides are removed, i.e. `custom_dir`, `extra_javascript` and `extra_css`
- [X] ... the documentation does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem

### Description

I've just noticed that my browsers based on Chromium no longer render a mermaid diagram that is residing inside a `??? note ""` block. Diagrams outside of the block are rendered fine as seen in the official documentation.

> Note: This problem doesn't seem to affect Firefox 96.

### Expected behaviour

Diagrams should be rendered _as normal_ regardless of being inside an admonition or details block.

### Actual behaviour

Browsers throw an error inside the console:

![grafik](https://user-images.githubusercontent.com/12431219/152781081-8410b252-c901-4003-a475-8b2e0f2e8dbd.png)

### Steps to reproduce

1. Create an `*.md` document that contains the following

```
??? note ""
    ``` mermaid
    graph LR
      A[Start] --> B{Error?};
      B -->|Yes| C[Hmm...];
      C --> D[Debug];
      D --> B;
      B ---->|No| E[Yay!];
    ```
```

2. Build the document.
3. Visit document with chromium-based browser.

### Package versions

- Python: `3.9.2`
- MkDocs: `1.2.3`
- Material: `8.1.10+insiders.4.8.1`

### Configuration

```yaml
site_name: My Docs
theme: material
markdown_extensions:
  - pymdownx.superfences:
      custom_fences:
        - name: mermaid
          class: mermaid
          format: !!python/name:pymdownx.superfences.fence_code_format
```

### System information

- Operating system: Debian 11, Windows 10
- Browser: Chrome 98, Edge 98

**Comment by squidfunk** [comment]
Thanks for reporting. Fixed in [`97f8233`](https://github.com/squidfunk/mkdocs-material-insiders/commit/97f82332868ff451d7f9998005f9dd79f1d5ebb6). The error is only happening for collapsed admonition elements. The reason is that `Element.innerText` does not return anything when elements are invisible, but `Element.textContent` seems to do the trick. Since Mermaid blocks are not syntax highlighted, it should be safe to switch to `Element.textContent`, as there's not markup for Mermaid to choke on.

**Comment by squidfunk** [comment]
Released as part of 8.1.11.
